### PR TITLE
1455 Multiple execute trade

### PR DESCRIPTION
**Pull request by namnm1991**

### CODE REVIEW

File: tradelogs/crawler.go [hunk 1]
```diff
@@ -233,24 +209,42 @@ func (crawler *Crawler) GetTradeLogs(fromBlock, toBlock *big.Int, timeout time.D
...
-			if result, err = updateTradeLogs(result, logItem, ts); err != nil {
+			err := updateTradeLog(&tradeLog, logItem)
+			if err != nil {
 				return result, err
 			}
+
+			// a tradeEvent indicate that all tradeLog information was fulfilled
+			// we add the tradeLog into result and prepare to fulfill an other
+			// tradeLog
+			if topic.Hex() == tradeEvent {
```
Comment (favadi) [inline tradelogs/crawler.go]: please add a debug log here with some basic trade log information in context.

File: tradelogs/crawler.go [hunk 2]
```diff
@@ -233,24 +209,42 @@ func (crawler *Crawler) GetTradeLogs(fromBlock, toBlock *big.Int, timeout time.D
...
+		if len(logItem.Topics) == 0 {
+			return result, errors.New("log item has no topic")
+		}
+
 		topic := logItem.Topics[0]
 		switch topic.Hex() {
 		case feeToWalletEvent, burnFeeEvent, etherReceivalEvent, tradeEvent:
-			// add logItem to result
-			if result, err = updateTradeLogs(result, logItem, ts); err != nil {
+			err := updateTradeLog(&tradeLog, logItem)
```
Comment (favadi) [inline tradelogs/crawler.go]: err is already declared above, also we should us `if xxx; err != nil {}` form.

File: tradelogs/crawler.go (hunk 1 above)
Comment (favadi) [reply tradelogs/crawler.go]: is it clearer if we move this logic to another case of switch statement? `case tradeEvent`?

File: tradelogs/common/types.go [hunk 3]
```diff
@@ -46,6 +48,8 @@ type TradeLog struct {
 
 	ETHUSDRate     float64 `json:"-"`
 	ETHUSDProvider string  `json:"-"`
+
+	Index uint `json:"index"`
```
Comment (favadi) [inline tradelogs/common/types.go]: please add a comment for this field, something like log index of in transactions events.

File: tradelogs/crawler.go [hunk 4]
```diff
@@ -116,80 +116,56 @@ func logDataToBurnFeeParams(data []byte) (ethereum.Address, ethereum.Hash, error
...
-	if !updateLastLog {
-		tradeLog = common.TradeLog{
-			BlockNumber:     logItem.BlockNumber,
-			TransactionHash: logItem.TxHash,
-			Timestamp:       ts,
-		}
-	}
-
-	switch logItem.Topics[0].Hex() {
+func updateTradeLog(tradeLog *common.TradeLog, eventLog types.Log) error {
```
Comment (favadi) [inline tradelogs/crawler.go]: I think it should be clearer if we can refactor this to a function that accept `[]types.Log` and output `[]TradeLog, error`.

File: tradelogs/common/types.go (hunk 3 above)
Comment (namnm1991) [reply tradelogs/common/types.go]: changed in 0ebe190eea0b12e76b479634fc5d0f2c2dc8e586

File: tradelogs/crawler.go (hunk 4 above)
Comment (namnm1991) [reply tradelogs/crawler.go]: added new function in fa5a5d3a2453f656e96278816a655531d10f1bc5

File: tradelogs/crawler.go (hunk 2 above)
Comment (namnm1991) [reply tradelogs/crawler.go]: changed in fa5a5d3a2453f656e96278816a655531d10f1bc5

File: tradelogs/crawler.go (hunk 1 above)
Comment (namnm1991) [reply tradelogs/crawler.go]: changed in fa5a5d3a2453f656e96278816a655531d10f1bc5 with log and seperated logic

**Pull request merged**